File: app/src/main/java/de/bahnhoefe/deutschlands/bahnhofsfotos/db/StationListAdapter.java
package de.bahnhoefe.deutschlands.bahnhofsfotos.db;

import android.content.Context;
import android.database.Cursor;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CursorAdapter;

import de.bahnhoefe.deutschlands.bahnhofsfotos.R;
import de.bahnhoefe.deutschlands.bahnhofsfotos.databinding.ItemStationBinding;
import de.bahnhoefe.deutschlands.bahnhofsfotos.util.Constants;

public class StationListAdapter extends CursorAdapter {
    private final LayoutInflater mInflater;

    public StationListAdapter(final Context context, final Cursor cursor, final int flags) {
        super(context, cursor, flags);
        mInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public View newView(final Context context, final Cursor cursor, final ViewGroup parent) {
        final var binding = ItemStationBinding.inflate(mInflater, parent, false);
        final var view = binding.getRoot();
        view.setTag(binding);
        return view;
    }

    @Override
    public void bindView(final View view, final Context context, final Cursor cursor) {
        //If you want to have zebra lines color effect uncomment below code
        if (cursor.getPosition() % 2 == 1) {
            view.setBackgroundResource(R.drawable.item_list_backgroundcolor);
        } else {
            view.setBackgroundResource(R.drawable.item_list_backgroundcolor2);
        }

        final var binding = (ItemStationBinding) view.getTag();
        binding.txtState.setText(cursor.getString(cursor.getColumnIndexOrThrow(Constants.STATIONS.COUNTRY)).concat(": ").concat(cursor.getString(cursor.getColumnIndexOrThrow(Constants.STATIONS.ID))));
        binding.txtStationName.setText(cursor.getString(cursor.getColumnIndexOrThrow(Constants.STATIONS.TITLE)));
        binding.hasPhoto.setVisibility(cursor.getString(cursor.getColumnIndexOrThrow(Constants.STATIONS.PHOTO_URL)) != null? View.VISIBLE : View.INVISIBLE);
    }

}

File: app/src/main/kotlin/de/bahnhoefe/deutschlands/bahnhofsfotos/db/StationListAdapter.kt
package de.bahnhoefe.deutschlands.bahnhofsfotos.db

import android.content.Context
import android.database.Cursor
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CursorAdapter
import de.bahnhoefe.deutschlands.bahnhofsfotos.R
import de.bahnhoefe.deutschlands.bahnhofsfotos.databinding.ItemStationBinding
import de.bahnhoefe.deutschlands.bahnhofsfotos.util.Constants.Stations

class StationListAdapter(context: Context, cursor: Cursor?, flags: Int) :
    CursorAdapter(context, cursor, flags) {
    private val mInflater =
        context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

    override fun newView(context: Context, cursor: Cursor, parent: ViewGroup): View {
        val binding = ItemStationBinding.inflate(mInflater, parent, false)
        val view = binding.root
        view.tag = binding
        return view
    }

    override fun bindView(view: View, context: Context, cursor: Cursor) {
        if (cursor.position % 2 == 1) {
            view.setBackgroundResource(R.drawable.item_list_backgroundcolor)
        } else {
            view.setBackgroundResource(R.drawable.item_list_backgroundcolor2)
        }
        (view.tag as ItemStationBinding).apply {
            txtStationKey.text = getStationkey(cursor)
            txtStationName.text = cursor.getString(cursor.getColumnIndexOrThrow(Stations.TITLE))
            hasPhoto.visibility =
                if (cursor.getString(cursor.getColumnIndexOrThrow(Stations.PHOTO_URL)) != null) View.VISIBLE else View.INVISIBLE
        }
    }

    private fun getStationkey(cursor: Cursor) =
        cursor.getString(cursor.getColumnIndexOrThrow(Stations.COUNTRY)) + ": " + cursor.getString(
            cursor.getColumnIndexOrThrow(Stations.ID)
        )
}